jcl/src/java.base/share/classes/java/lang/StringBuilder.java

```diff
@@ -664,7 +664,7 @@ public StringBuilder append (String string) {
 			ensureCapacityImpl(newLength);
 		}
 		
-		string.getChars(0, stringLength, value, currentLength);
+		string.getCharsImpl(0, stringLength, value, currentLength);
```

Comment:
Why is this `getChars` using the fast path when the other call to `getChars` in this method doesn't?  If this one is doing something special, please indicate why

Reply:
Also, is this change actually noticeable in the ~1%?  Is it still ~1% without this extra fast path?  If so, I'm inclined to avoid the code complexity here unless this is a measurable part of the improvement.

Reply:
-- Both cases should use `getCharsImpl` / `getCharsNoBoundChecks` as both have validated their bound checks.
-- The 1% measurement on Open Liberty ACEMEAIR benchmark include this fast path for `getChars` along the `StringBuilder.append(String)` chain of calls. It was attributed to both removing the bound checks by calling `getCharsImpl` and using the JIT recognized decompressedArrayCopy to further remove array copy bound checks,

Reply:
Moving this to the main discussion thread so it's easier to follow the discussion....